src/app.js

```diff
@@ -2,12 +2,252 @@
+    event.preventDefault();
+    tripList.forEach((trip) => {
+      $tripList.append(tripTemplate(trip.attributes));
+    });
+  },
+  showTrip(id) {
+    const $showTrip = $('#show_trip');
+    $showTrip.empty();
+    $showTrip.show();
+    const trip = new Trip({id: id});
+    const resForm = `<h2>Reserve Your Spot</h2>
+    <form id="reserve-trip-form" "action="https://trektravel.herokuapp.com/trips/${trip.id}/reservations" method="post">
+
+    <label for="person">Name</label>
+    <input type="text" name="person" placeholder="Enter Name"></input>
+
+    <label for="age">Age</label>
+    <input type="number" name="age" placeholder="Enter Age"></input>
+
+    <label for="email">Email</label>
+    <input type="text" name="email" placeholder="Enter Email"></input>
+
+    <input type="hidden" value="${trip.id}" name="trip_id" />
+
+    <input id="submitResButton" type="submit" value="Reserve it!" class="button"></input>
+    </form>`
+    trip.fetch({}).done(() => {
+      $showTrip.append(resForm);
+      $showTrip.append(showTemplate(trip.attributes));
+    });
+
+  },
+  finalizeReservation(event) {
+    event.preventDefault();
+    const url = $(this).attr('action'); // Retrieve the action from the form
+    const resData = {};
+    reservationFields.forEach( (field) => {
+      if (field === 'person') {
+        const val = $(`input[name='person']`).val();
+      }
+      const val = $(`input[name=${field}]`).val();
+      if (val !== '') {
+        resData[field] = val;
+      }
+    });
+    resData['name'] = $(`input[name='person']`).val();
+    delete resData['person'];
+    const res = new Reservation(resData);
+    if (res.isValid()) {
+      res.save({}, {
+        success: events.succesfulResSave,
+        error: events.failedResSave,
+    });
+  } else {
+    // getting here means there were client-side validation errors reported
+    $('#message').empty();
+    let messageObj = res.validationError;
+    for(let messageType in messageObj) {
+      messageObj[messageType].forEach((message) => {
+        $('#message').append($(`<p>${message}</p>`));
+      })
+    }
+    $('#message').show();
+    $('#message').addClass('errors');
+    $('#message').delay(2000).hide(1);
+    console.log("RESERVATION VALIDATION ERRORS!!")
+  }
+},
+succesfulResSave() {
+  $('#show_trip').hide();
+  $('#message').show();
+  $('#message').empty();
+  $('#message').removeClass('errors');
+  $('#message').html('<p> Trip Reserved! </p>');
+  $('#message').delay(1000).hide(1);
+},
+failedResSave() {
+  $('#message').empty();
+  $('#message').html('<p>Reserving Trip Failed</p>');
+  $('#message').delay(2000).hide(1);
+},
+addTrip(event) {
+  event.preventDefault();
+  const tripData = {};
+  newTripFields.forEach( (field) => {
+    const val = $(`input[name=${field}]`).val();
+    if (val !== '') {
+      tripData[field] = val;
+    }
+  });
+  const trip = new Trip(tripData);
+  if (trip.isValid()) {
+    tripList.add(trip);
```

Comment:
This is going to trigger an error because the `id` is assigned by the API, but you haven't recieved any data from the API yet.  A better way would be to add the trip after fetching the trip's data from the API (after a successful save).